jobs/validate/nagios-spec.yml

```diff
@@ -0,0 +1,82 @@
+plan:
+  - &BASE_JOB
+    env:
+      - SNAP_VERSION=1.17/edge
+      - JUJU_DEPLOY_BUNDLE=cs:~containers/charmed-kubernetes
+      - JUJU_DEPLOY_CHANNEL=edge
+      - JUJU_CLOUD=aws/us-east-2
+      - JUJU_CONTROLLER=validate-ck-nagios
+      - JUJU_MODEL=validate-nagios
+    before-script:
+      - juju:
+          cloud: $JUJU_CLOUD
+          controller: $JUJU_CONTROLLER
+          model: $JUJU_MODEL
+          bootstrap:
+            constraints: "arch=amd64"
+            debug: no
+            replace-controller: yes
+            model-default:
+              - test-mode=true
+              - resource-tags=owner=k8sci
+          deploy:
+            reuse: no
+            bundle: $JUJU_DEPLOY_BUNDLE
+            overlay: |
+              applications:
+                kubernetes-master:
+                  options:
+                    channel: $SNAP_VERSION
+                kubernetes-worker:
+                  options:
+                    channel: $SNAP_VERSION
+            wait: yes
+            timeout: 7200
+            channel: $JUJU_DEPLOY_CHANNEL
+
+    script:
+      - |
+        #!/bin/bash
+        set -x
+        pytest $INTEGRATION_TEST_PATH/test_nagios.py::test_nagios \
+           --cloud $JUJU_CLOUD \
+           --model $JUJU_MODEL \
+           --controller $JUJU_CONTROLLER 2>&1
+
+    after-script:
+      - runner:
+          script: |
+            #!/bin/bash
+            set -x
+            wget https://raw.githubusercontent.com/juju-solutions/cdk-field-agent/master/collect.py
+            python3 collect.py -m $JUJU_CONTROLLER:$JUJU_MODEL
+            python3 jobs/infra/collect-debug.py push 'cdk_field_agent' results*.tar.gz && rm -rf results*.tar.gz
+            python3 jobs/infra/collect-debug.py push 'build_log' ogc.log
+            python3 jobs/infra/collect-debug.py push 'metadata' metadata.json
+            python3 jobs/infra/collect-debug.py push 'job_result' *job.json
+            python3 jobs/infra/collect-debug.py set-key 'snap_version' "$SNAP_VERSION"
+            python3 jobs/infra/collect-debug.py save-meta metadata.json
+            juju-crashdump -a debug-layer -a config -m "$JUJU_CONTROLLER:$JUJU_MODEL" || true
+            python3 jobs/infra/collect-debug.py push 'results_file' juju-crashdump* || true
+
```

Comment:
You can re-use 'spec-helpers' like https://github.com/charmed-kubernetes/jenkins/blob/master/jobs/validate/spec.yml#L15-L17